src/lib/containers/download_list/DownloadListTable.tsx

```diff
@@ -0,0 +1,264 @@
+  // https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables
+  let [data, setData] = useState<DownloadListTableData>({
+    references: undefined,
+    batchFileResult: undefined,
+    downloadList: undefined,
+  })
+  // https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables
+  let [isLoading, setIsLoading] = useState<LoadingState>(true)
+  const { token } = props
+  const { references, batchFileResult, downloadList } = data
+  const requestedFiles =
+    (batchFileResult && batchFileResult.requestedFiles) || []
+  const ownerIds: any = requestedFiles
+    .map(el => el.fileHandle && el.fileHandle.createdBy)
+    .filter(el => el)
+  let userProfiles = useGetProfiles({ ids: ownerIds, token })
+
+  useEffect(() => {
+    fetchData(token)
+  }, [token])
+
+  const fetchData = async (token: string | undefined) => {
+    if (!token) {
+      // doesn't make sense with anonymous user!
+      return
+    }
+    try {
+      setIsLoading(true)
+      const downloadList = await getDownloadList(token)
+      const { filesToDownload } = downloadList
+      if (filesToDownload.length === 0) {
+        setData({
+          downloadList,
+        })
+        setIsLoading(false)
+        return
+      }
+      const referenceCall: Reference[] = filesToDownload.map(el => {
+        return { targetId: el.fileHandleId }
+      })
+      // entity header is used to get the names of the files that the user
+      // doesn't have access to
+      const references = await getEntityHeader(referenceCall, token)
+      const batchFileRequest: BatchFileRequest = {
+        requestedFiles: filesToDownload,
+        includeFileHandles: true,
+        includePreSignedURLs: false,
+        includePreviewPreSignedURLs: false,
+      }
+      // batch file result gives FilesHandle for the files the user can download
+      // which has additional metadata - createdBy, numBytes, etc.
+      const batchFileResult = await getFiles(batchFileRequest, token)
+      setData({
+        references,
+        batchFileResult,
+        downloadList,
+      })
+      setIsLoading(false)
+    } catch (e) {
+      console.error('Error in DownloadList API call : ', e)
+    }
+  }
+
+  const clearDownloadList = (
+    _event: React.SyntheticEvent<HTMLButtonElement>,
+  ) => {
+    setIsLoading(true)
+    deleteDownloadList(token)
+      .then(() => {
+        setIsLoading(false)
+        setData({
+          downloadList: undefined,
+        })
+      })
+      .catch(err => {
+        console.error('Error on clearing download list: ', err)
+      })
+  }
+
+  const deleteFileFromList = (
+    fileHandleId: string,
+    associateObjectId: string,
+  ) => {
+    const list: FileHandleAssociation[] = [
+      {
+        fileHandleId,
+        associateObjectId,
+        associateObjectType: FileHandleAssociateType.FileEntity,
+      },
+    ]
+    setIsLoading(true)
+    deleteDownloadListFiles(list, token)
```

Comment:
same quesiton about sync/await and also do we need  `setIsLoading(false)` somewhere?